i2p: fix build on Linux

danielnachun opened the pull request:
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

danielnachun commented:
I supposed the file name here should have been an indicator that this would run afoul of the universal binary audit on macOS.  We'll have to see what options are available to build architecture specific binaries.

cho-m commented:
`i2p` currently uses `ppc64 + x86_64` universal binaries, so in current state it can't run on Apple Silicon. `arm64_big_sur` bottle may be broken without Rosetta.

The workaround in other formulae has been to use `java-service-wrapper` formula and overwrite the existing copy.

---

Note that similar to many other Java formulae and Maven packages, the libraries/binaries aren't built but just copied over from official releases. Working around this is often tricky based on Java/Maven's common distribution approach (similar problem with Node/NPM)

danielnachun commented:
This went in a very different direction than I expected but I think it's all fixed now.  It turns out the Launchpad upstream is no longer being updated because development moved to GitLab and releases now come from their own website instead.  

I've updated the formula to use the latest release (hence the change in PR name and commit message), and I basically had to refactor the whole formula.  I took inspiration from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=i2p, where it builds from source instead of using the JAR installer.  As suggested by @cho-m, I removed the vendored `java-service-wrapper` and replaced it with the brewed version instead.
